langkit/support/langkit_support-token_data_handlers.adb

```diff
@@ -681,16 +688,117 @@ package body Langkit_Support.Token_Data_Handlers is
    begin
       --  Return slice from...
       return
-        TDH.Source_Buffer (
-         --  The first character in the requested line
-         TDH.Lines_Starts.Get (Line_Number)
+        TDH.Source_Buffer
+          (
+           --  The first character in the requested line
+           TDH.Lines_Starts.Get (Line_Number)
+
+           ..
 
-         ..
+           --  The character before the LF that precedes the first character of
+           --  the next line.
+             TDH.Lines_Starts.Get (Line_Number + 1) - 2
+          );
 
-         --  The character before the LF that precedes the first character of
-         --  the next line.
-           TDH.Lines_Starts.Get (Line_Number + 1) - 2
-        );
    end Get_Line;
 
+   --------------
+   -- Get_Sloc --
+   --------------
+
+   function Get_Sloc
+     (TDH : Token_Data_Handler; Offset : Natural) return Source_Location
+   is
+      function Compare
+        (Sought      : Positive;
+         Dummy_Index : Positive;
+         Line_Start  : Positive) return Relative_Position
+      is
+        (if Sought > Line_Start then After
+         elsif Sought = Line_Start then Inside
+         else Before);
+
+      function Get_Line_Index is new Floor (Positive, Index_Vectors);
+
+      Column      : Natural := 0;
+
+      Tab_Stop : Natural renames TDH.Tab_Stop.all;
+   begin
+      --  Allow 0 as an offset because it's a common value when the text buffer
+      --  is empty: in that case just return a null source location.
+      if Offset = 0 then
+         return No_Source_Location;
+      end if;
+
+      declare
+         Line_Index  : constant Positive :=
+           Get_Line_Index (Offset, TDH.Lines_Starts);
+         Line_Offset : constant Positive := TDH.Lines_Starts.Get (Line_Index);
+      begin
+         --  Allow a sloc pointing at the EOL char (hence the + 1)
+         if Offset > TDH.Source_Buffer'Last + 1 then
+            raise Constraint_Error with "out of bound access";
+         end if;
+
+         --  Make horizontal tabulations move by stride of 8 columns, as
+         --  usually implemented in code editors.
```

Comment:
s/8/Tab_Stop/

Reply:
You're catching typos in your own comments that I moved around :P